### PR TITLE
v2: fix several panics

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Ref #12842 

What's Changed:

```commit-message
Perf context is disabled for now as we don't have shared kv engine.
And fix region info access panic by filter out uninitialized role change.
There are also several other fixes.
```

### Check List 

Tests 

- Unit test
- Integration test
- Manual test
    tiup playground

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/config/mod.rs [hunk 1]
```diff
@@ -2988,6 +2988,10 @@ impl TikvConfig {
                 .to_owned();
         }
 
+        if self.storage.engine == EngineType::RaftKv2 {
```
Comment (tabokie) [inline src/config/mod.rs:2991]: Why?

File: components/server/src/server2.rs [hunk 2]
```diff
@@ -1539,14 +1542,18 @@ impl<EK: KvEngine, ER: RaftEngine> EngineMetricsManager<EK, ER> {
...
         self.tablet_registry
             .for_each_opened_tablet(|_, db: &mut CachedTablet<EK>| {
                 if let Some(db) = db.latest() {
                     reporter.collect(db);
+                    collected = true;
                 }
                 true
             });
-        reporter.flush();
+        if collected {
```
Comment (tabokie) [inline components/server/src/server2.rs:1554]: Why? Flushing empty should be a no-op.

File: src/config/mod.rs (hunk 1 above)
Comment (BusyJay) [reply src/config/mod.rs:2991]: raftstore 2 always use async write, and the configuration is by default 0.

File: components/server/src/server2.rs (hunk 2 above)
Comment (BusyJay) [reply components/server/src/server2.rs:1554]: Because arrays are initialized in `collect`. If collect has never been called, flush will panic.

**Review by tabokie — APPROVED** [review]
Rest LG

**Review by tonyxuqqi — APPROVED** [review]

**Pull request merged**